account/tests.py

```diff
@@ -2,39 +2,62 @@
 import re
 from account.models import validate_gov_id, validate_phone, validate_pin_code
 from django.core.exceptions import ValidationError
+from account.factories import CustomUserFactory
+import django
 
+class TestCustomUserValidators(django.test.TestCase):
+    
+    def test_valid_phone_number(self):
+        user = CustomUserFactory.create(phone='3336647')
+        
+        self.assertTrue(user.phone.isdigit())
+        self.assertEqual(len(user.phone), 7)
 
-class TestCustomUserValidators(unittest.TestCase):
-
-    #test for phone validation
-     def test_phone_validator(self):
+    def test_invalid_phone_number(self):
+        user = CustomUserFactory.create(phone='589929a')
+        with self.assertRaises(ValidationError):
+            validate_phone(user.phone)
 
-        self.assertEqual('5899298', '5899298')
 
-        with self.assertRaises(ValidationError):
-            validate_phone('58992989')
-        with self.assertRaises(ValidationError):
-            validate_phone('5899298a')
-    
-    #test for gov id validation
-     def test_govid_validator(self):
-        self.assertEqual('12345678','12345678')
-        self.assertEqual('1234567', '1234567')
-        self.assertEqual('123456', '123456')
-        self.assertEqual('12345', '12345')
+    def test_valid_pin_code(self):
         
+        user = CustomUserFactory.create(pin_code = '5zxu4sd')
+
+        self.assertTrue(user.pin_code.isalnum())
+        self.assertEqual(len(user.pin_code), 7)
+
+    def test_invalid_pin_code(self):
+        user = CustomUserFactory.create(pin_code='5zxu4prf')
         with self.assertRaises(ValidationError):
```

Comment:
Does this work? I believe create should give error, as clean method is called, no? So, ValidationError happens before you assert it

Reply:
yes, test run successfully. if i change the pin_code value to be correct, it fails, and output AssertionError: ValidationError not raised